Identify why test_mp_checkout is failing and how to fix it

tests/test_broker.py test_mp_checkout is currently failing since the multiprocessing queue hangs on get.
This works perfectly fine when using the AnsibleTower provider, but this test uses the TestProvider.

Alternative solutions tried, but failed are:
- Use starmap to get around a queue
- Introduce a time delay in TestProvider's test_action method
- Use a multiprocessing manager queue
- Use a multiprocessing manager list
- Make the Host class instances pickleable to be compatible with above

The process child processes are zombies:
```
jhenner  3347498  3.2  0.5 296956 42672 pts/1    S    01:50   0:00 /home/jhenner/work/sat/broker/venv/bin/python venv/bin/py.test tests -v
jhenner  3347507  0.0  0.0      0     0 pts/1    Z    01:50   0:00 [py.test] <defunct>
jhenner  3347508  0.0  0.0      0     0 pts/1    Z    01:50   0:00 [py.test] <defunct>
```

The parent hangs when dealing with the Queue
```
(gdb) py-bt
Traceback (most recent call first):
  <built-in method read of module object at remote 0x7fd1114b9680>
  File "/usr/lib64/python3.9/multiprocessing/connection.py", line 384, in _recv
    chunk = read(handle, remaining)
  File "/usr/lib64/python3.9/multiprocessing/connection.py", line 419, in _recv_bytes
    buf = self._recv(4)
  File "/usr/lib64/python3.9/multiprocessing/connection.py", line 221, in recv_bytes
    buf = self._recv_bytes(maxlength)
  File "/usr/lib64/python3.9/multiprocessing/queues.py", line 103, in get
    res = self._recv_bytes()
  File "/home/jhenner/work/sat/broker/broker/broker.py", line 75, in _mp_split
    result.append(mp_queue.get())
  File "/home/jhenner/work/sat/broker/broker/broker.py", line 27, in wrapper
    return vmb_inst._mp_split(vmb_inst.checkout, count, **vmb_inst._kwargs)
  File "/home/jhenner/work/sat/broker/tests/test_broker.py", line 50, in test_mp_checkout
```

I started an attempt  to fix this using the concurrent.futures which I find easy to use, foolproof and enough for many situations. Right now I am solving it's complain that 
```
 E           RuntimeError: Queue objects should only be shared between processes through inheritance
```

--- a/broker/providers/test_provider.py
+++ b/broker/providers/test_provider.py
@@ -17,8 +17,25 @@ HOST_PROPERTIES = {
 
 class TestProvider(Provider):
     def __init__(self, **kwargs):
-        self.config = settings.TESTPROVIDER
+        # self.config = settings.TESTPROVIDER
+
+        # Enabling the line above leads to problems to pickle in
+        # tests/test_broker.py::test_mp_checkout
+        # that resulted in deadlock. This was quite hard to find problem as the tracebacks and
+        # errors I was getting were not helpful:
+        # Traceback (most recent call last):
+        #   File "/usr/lib64/python3.9/multiprocessing/queues.py", line 245, in _feed
+        #     obj = _ForkingPickler.dumps(obj)
+        #   File "/usr/lib64/python3.9/multiprocessing/reduction.py", line 51, in dumps
+        #     cls(buf, protocol).dump(obj)
+        # TypeError: cannot pickle 'module' object
+        #
+        # The root cause was tracked down with help of
+        # https://stackoverflow.com/a/59832602/1950100
+
+
         # self.__dict__.update(kwargs)
+        pass
 
     def _host_release(self):
         caller_host = inspect.stack()[1][0].f_locals["host"]

--- a/broker/providers/test_provider.py
+++ b/broker/providers/test_provider.py
@@ -17,25 +17,9 @@ HOST_PROPERTIES = {
 
 class TestProvider(Provider):
     def __init__(self, **kwargs):
-        # self.config = settings.TESTPROVIDER
-
-        # Enabling the line above leads to problems to pickle in
-        # tests/test_broker.py::test_mp_checkout
-        # that resulted in deadlock. This was quite hard to find problem as the tracebacks and
-        # errors I was getting were not helpful:
-        # Traceback (most recent call last):
-        #   File "/usr/lib64/python3.9/multiprocessing/queues.py", line 245, in _feed
-        #     obj = _ForkingPickler.dumps(obj)
-        #   File "/usr/lib64/python3.9/multiprocessing/reduction.py", line 51, in dumps
-        #     cls(buf, protocol).dump(obj)
-        # TypeError: cannot pickle 'module' object
-        #
-        # The root cause was tracked down with help of
-        # https://stackoverflow.com/a/59832602/1950100
-
+        self.config = settings.TESTPROVIDER.config_value
 
         # self.__dict__.update(kwargs)
-        pass
 
     def _host_release(self):
         caller_host = inspect.stack()[1][0].f_locals["host"]

--- a/tests/test_broker.py
+++ b/tests/test_broker.py
@@ -42,11 +42,9 @@ def test_broker_e2e():
     broker_inst.checkin()
     assert len(broker_inst._hosts) == 0
 
-# Deprecated for now, as I'm unsure why this is failing
-# This functionality works with AnsibleTower
-# def test_mp_checkout():
-#     """Test that broker can checkout multiple hosts using multiprocessing"""
-#     broker_inst = broker.VMBroker(nick="test_nick", _count=2)
-#     broker_inst.checkout()
-#     assert len(broker_inst._hosts) == 2
-#     broker_inst.checkin()
+def test_mp_checkout():
+    """Test that broker can checkout multiple hosts using multiprocessing"""
+    broker_inst = broker.VMBroker(nick="test_nick", _count=2)
+    broker_inst.checkout()
+    assert len(broker_inst._hosts) == 2
+    broker_inst.checkin()

